wagtail/search/utils.py

```diff
@@ -38,3 +40,43 @@ def separate_filters_from_query(query_string):
     query_string = re.sub(filters_regexp, '', query_string).strip()
 
     return filters, query_string
+
+
+def parse_query_string(query_string, operator=None, zero_terms=MATCH_NONE):
+    """
+    This takes a query string typed in by a user and extracts the following:
+
+     - Quoted terms (for phrase search)
+     - Filters
+
+    For example, the following query:
+
+      `hello "this is a phrase" live:true` would be parsed into:
+
+    filters: {'live': 'true'}
+    tokens: [('hello', False), ('this is a phrase', True)]
```

Comment:
This should be a SearchQuery expression... I was puzzled where this tuple notation in d53affb623b7769ce612558b7b9377d2477f3de7 arose from, until I saw the fixup commit that showed it was a mistake :-)

```suggestion
    tokens: And([PlainText('hello'), Phrase('this is a phrase')])
```